Fix android build

cajus opened the pull request:
closes https://github.com/bincrafters/community/issues/710 
These changes helped me when building Qt for Android. Note that not all requirements seem to be there in *bincrafters* - i.e. harfbuzz needs to be fixed. And an android SDK package is needed as a dependency in the profile.

I've opened the PR for sharing and further discussion as a follow up of https://github.com/bincrafters/community/issues/710.

cajus committed:
Prepare compilation for android

Note that this will not work yet, because there are some dependencies
that do not build without patches (i.e. harfbuzz). Additionally there
needs to be some android SDK package around - which doesn't exist within
bincrafters. I'm using https://github.com/Tereius/conan-android-sdk in
the meantime.

cajus committed:
Remove android-sdk for now

ericLemanissier commented on conanfile.py:
```diff
@@ -534,9 +547,10 @@ def _getenvpath(var):
-            args += ['QMAKE_CXX="' + value + '"',
-                     'QMAKE_LINK="' + value + '"',
-                     'QMAKE_LINK_SHLIB="' + value + '"']
+            args += ['QMAKE_CXX="' + value + '"']
```
Is this change still needed with the `-nostdlib++` patch below ?

cajus replied:
You're right. That's a left over.

cajus committed:
Remove pre -nostdlib++ left over

SSE4 commented on conanfile.py:
```diff
@@ -299,6 +301,9 @@ def source(self):
             self.run("wget -qO- %s.tar.xz | tar -xJ " % url)
         shutil.move("qt-everywhere-src-%s" % self.version, "qt5")
 
+        for patch in glob.glob("patches/*.diff"):
```
please rename patch to start with `0001-` and add `sorted(glob.glob(...))` here.
it's important for the future that patches apply in the right and consistent order.

cajus committed:
Sort patches

Very effective, if there's just one, but lets see what's happening
in the future :-)

ericLemanissier commented:
@cajus thanks a lot for the PR. Did you submit the patch to Qt ?

cajus commented:
Indirectly. I've sent it to a Qt guy I know, but I didn't open a change there. The question is: how backwards compatible is the contribution for the project in general.

ericLemanissier commented:
This recipe is now in the Conan Center Index.

Please migrate to the recipe version there.
If this is still an issue, please create a new pull request in the Conan Center Index repository.

Thanks!

https://conan.io/center/qt
